### PR TITLE
refactor(notification): extract siteUrl cacluation logic into util

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:

In this PR, we extract the calculation logic of odc site url into a separated component `SiteUrlResolver` for reuse.
We also use `JdbcTemplate` to replace mybatic in `SystemConfigService`.

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/config/SystemConfigDAO.java [hunk 1]
```diff
@@ -17,10 +17,46 @@
...
+    public List<SystemConfigEntity> queryByKeyPrefix(String keyPrefix) {
+        PreConditions.notNull(keyPrefix, "keyPrefix");
+        String sql = "SELECT `application`, `profile`, `key`, `value`, `create_time`, `update_time`, `description` "
+                + "FROM `config_system_configuration` "
+                + "WHERE `application`='odc' AND `profile`='default' AND `label`='master' AND `key` LIKE ?";
+        return jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(SystemConfigEntity.class), keyPrefix + "%");
+    }
+
+    public SystemConfigEntity queryByKey(String key) {
+        PreConditions.notEmpty(key, "ket");
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/metadb/config/SystemConfigDAO.java:42]: typo: ket

**Review by yizhouxw — COMMENTED** [review]
👍

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/config/SystemConfigDAO.java (hunk 1 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/config/SystemConfigDAO.java]: done~

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]

**Pull request merged**